### ISSUE TITLE
diffJson with dates, returns empty curly braces 

### ISSUE THREAD

**Issue by dr-dimitru** [kpdecker/jsdiff#102]
Example:
```js
var after = {
  "fingerprint" : "string1",
  "valid_to" : new Date("2016-08-14T23:59:59Z"),
  "valid_from" : new Date("2016-02-12T00:00:00Z")
};

var before = {
  "fingerprint" : "string 2",
  "valid_from" : new Date("2016-02-12T00:00:00Z"),
  "valid_to" : new Date("2016-02-20T03:58:53Z")
};

JsDiff.diffJson(before, after);
```

Result:
```console
[
  {
    "count": 1,
    "value": "{\n"
  },
  {
    "count": 1,
    "removed": true,
    "value": "  \"fingerprint\": \"string 2\",\n"
  },
  {
    "count": 1,
    "added": true,
    "value": "  \"fingerprint\": \"string1\",\n"
  },
  {
    "count": 3,
    "value": "  \"valid_from\": {},\n  \"valid_to\": {}\n}"
  }
]
```

Expected result:
```console
[
  {
    "count": 1,
    "value": "{\n"
  },
  {
    "count": 1,
    "removed": true,
    "value": "  \"fingerprint\": \"string 2\",\n"
  },
  {
    "count": 2,
    "added": true,
    "value": "  \"fingerprint\": \"string 1\",\n  \"valid_to\": \"2016-11-20T03:58:53.000Z\",\n"
  },
  {
    "count": 1,
    "value": "  \"valid_from\": \"2015-11-20T01:36:42.000Z\",\n"
  },
  {
    "count": 1,
    "removed": true,
    "value": "  \"valid_to\": \"2016-02-14T03:58:53.000Z\"\n"
  },
  {
    "count": 1,
    "value": "}"
  }
]
```

**Comment by dr-dimitru** [comment]
Maybe I'm missing something...

**Comment by kpdecker** [comment]
Date's can not be expressed in json, unless converted to a string or other primitive format. This is operating as expected, unfortunately.

**Comment by dr-dimitru** [comment]
Sorry, but I'm emphatically disagree, see next test:

```javascript
var obj = {str: 'asd', date: new Date()};
JSON.stringify(obj);
// result:
// "{\"str\":\"asd\",\"date\":\"2016-02-24T10:41:40.220Z\"}"
```

**Comment by kpdecker** [comment]
My comment that `Date` is not supported in JSON stands, but we should add some sort of `toString` operation to our JSON serializer to match the failover pattern of the standard serializer.

**Comment by dr-dimitru** [comment]
@kpdecker thank you. It should call `toString` on all objects, as it may be some custom class, what actually is happiness inside `JSON.stringify()` and objects without `toString` method returns as `[Object object]`

**Comment by kpdecker** [comment]
Released in 2.2.2
